Lib/http/client.py

```diff
@@ -1171,6 +1171,12 @@ def putrequest(self, method, url, skip_host=False,
                         netloc_enc = netloc.encode("ascii")
                     except UnicodeEncodeError:
                         netloc_enc = netloc.encode("idna")
+
+                    # remove interface scope from IPv6 address
+                    # when used as Host header
+                    if "%" in netloc:
```

Comment:
rather than checking netloc and modifying netloc_enc (which makes assumptions about the encoding - granted they're accurate so the logic works), how about doing this using the partition method for a single scan / split operation?

```python
netloc_enc, percent, unused_iface = netloc_enc.partition(b"%")
if percent:
    netloc_enc += b"]"
```

do similar with host_enc below.

There is technically a logic flaw in the current code, it's using netloc.find as an index into the encoded netloc_enc value.  (same with host vs host_enc).  the stripping should be done more directly from the same thing used for find.  partition will do that.

As this is done in two places, it'd be nice to turn it into a private helper function, put that logic in a top level `def _strip_ipv6_iface(enc_name: bytes) -> bytes:` function perhaps?

Reply:
hi @gpshead 
thank you for your review and your suggestion, which sounds very reasonable ... i've applied it now